### PR TITLE
automake: do not include CFLAGS in CPPFLAGS

### PR DESCRIPTION
**Pull request by rbuj**
Test:
```shell
$ gcc --version | head -n 1
gcc (GCC) 10.0.1 20200123 (Red Hat 10.0.1-0.5)
$ CFLAGS="--analyzer" CPPFLAGS="-DHELLO_WORLD" ./autogen.sh --prefix=/usr && make V=1 &> build.log
$ grep analyzer build.log | wc -l
62
$ grep HELLO_WORLD build.log | wc -l
45
```

### CODE REVIEW

**Comment by raveit65** [comment]
Looks like `CFLAGS="--analyzer"` only works with gcc10, i am using 9.2 here with f30.
```
[rave@mother caja-extensions-dist]$ gcc --version | head -n 1
gcc (GCC) 9.2.1 20190827 (Red Hat 9.2.1-1)

configure:3480: gcc --analyzer -DHELLO_WORLD  conftest.c  >&5
gcc: error: unrecognized command line option '--analyzer'
```

**Comment by raveit65** [comment]
Patch hangs
```
+ /usr/bin/patch -p1 -s --fuzz=0 --no-backup-if-mismatch
1 out of 1 hunk FAILED -- saving rejects to file sendto/Makefile.am.rej
```
OK first , the other PRs from caja-send-to needs to be merged.

**Comment by b4n** [comment]
I'm really not sure what this is supposed to achieve, but it doesn't look like it's doing what the title is suggesting.

Also, you don't have to explicitly list `$(CFLAGS)` or such in the Makefile.ams unless you're directly running a command that should use it; but Automake automatically includes it in the compiler commands.

What you have to do is include `$(AM_CFLAGS)` if you in target-specific `_CFLAGS` if you need its content.

See https://www.gnu.org/software/automake/manual/automake.html#Flag-Variables-Ordering

**Comment by rbuj** [comment]
@b4n done, the extra flags have been defined in separate variables, like automake recommends without overriding the precedence order.

**Comment by raveit65** [comment]
@b4n 
Is this Ok for you?

File: sendto/plugins/caja-burn/Makefile.am [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
+plugindir = $(libdir)/caja-sendto/plugins
 
 plugin_LTLIBRARIES = libnstburn.la
 
 libnstburn_la_SOURCES = caja-burn.c
+libnstburn_la_CPPFLAGS =			\
+	$(AM_CPPFLAGS)				\
+	-DDATADIR=\"$(datadir)\"		\
+	-DICONDIR=\"$(icondir)\"		\
+	-I$(srcdir)				\
```
Comment (cwendling) [inline sendto/plugins/caja-burn/Makefile.am]: Is that actually useful?  I would be surprised you actually have to do that

**Comment by cwendling** [comment]
OK, I had a look, and while it's a little too many changes at once for telling whether they all are good and nothing have been missed, the concept is good (and I now understand the PR's title): not mix what should go in `CFLAGS` and what should go in `CPPFLAGS`.  To that regard, it looks good :+1:

I can't actually test all what this has touched, at least not right now, but it looks OK and seems to build correctly with no new warnings.

File: sendto/plugins/caja-burn/Makefile.am (hunk 1 above)
Comment (rbuj) [reply sendto/plugins/caja-burn/Makefile.am]:
```shell
$ find . -type f -name Makefile.am -exec sed -i '/-I\$(srcdir)/d' {} \;
$ ./autogen.sh --prefix=/usr
$ make
<cut>
caja-burn.c:27:10: fatal error: nst-common.h: No such file or directory
   27 | #include "nst-common.h"
      |          ^~~~~~~~~~~~~~
compilation terminated.
<cut>
```

**Pull request merged**